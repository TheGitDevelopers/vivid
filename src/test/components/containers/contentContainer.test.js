import React from 'react';
import renderer from 'react-test-renderer';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { Text } from 'react-native';
import Container from '../../../components/common/containers/content-container/container';

describe('ContentContainer tests:', () => {
  /* Test mockups */
  configure({ adapter: new Adapter() });

  const ChildComponent = () => <Text>Content</Text>;
  const tree = renderer
    .create(<Container><ChildComponent /></Container>);

  const wrapper = shallow(<Container><ChildComponent /></Container>);

  /* All component tests should be listed here */
  it('renders correctly', () => {
    expect(tree.toJSON()).toMatchSnapshot();
  });
  it('renders child correctly', () => {
    expect(wrapper.children()).toHaveLength(1);
  });
});
