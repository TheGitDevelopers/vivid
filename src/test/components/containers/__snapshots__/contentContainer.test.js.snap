// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContentContainer tests: renders correctly 1`] = `
<View
  accessible={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "flex": 1,
    }
  }
>
  <RCTScrollView
    contentContainerStyle={
      Object {
        "alignItems": "center",
        "justifyContent": "flex-start",
        "paddingHorizontal": 30,
        "paddingVertical": 50,
      }
    }
    nestedScrollEnabled={true}
  >
    <View>
      <Text>
        Content
      </Text>
    </View>
  </RCTScrollView>
</View>
`;
